### PR TITLE
ruff: replace isort with ruff +TPU

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Isort is anyway using black style so we can just stick with black and validate it with Ruff

In several cases, Isoert failed to remove unused import and Ruff seems to perform better

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

cc @carmocca @borda @justusschock @awaelchli

### CODE REVIEW

**Review by carmocca — COMMENTED** [review]
Perhaps you need to find a ruff config that is equivalent to our current isort config

File: examples/app/dag/app.py [hunk 1]
```diff
@@ -1,15 +1,14 @@
...
 import numpy as np
 import pandas as pd
-from sklearn import datasets
-from sklearn.metrics import mean_squared_error
-
-import lightning as L
 from lightning.app.components import TracerPythonScript
 from lightning.app.storage import Payload
 from lightning.app.structures import Dict, List
+from sklearn import datasets
```
[the remark this answers is not in the document]
Comment (Borda) [reply examples/app/dag/app.py:10]: yes and it is such in the package, space removal is in examples and tests

[the remark this answers is not in the document]
Comment (carmocca) [reply examples/app/dag/app.py:10]: Why cant examples and tests stay as they are?

[the remark this answers is not in the document]
Comment (Borda) [reply examples/app/dag/app.py:10]: because in these cases `lightning` is regularly installed package
What exactly do you dislike except it is a change?

[the remark this answers is not in the document]
Comment (carmocca) [reply examples/app/dag/app.py:10]: I would say that ideally the tests imports wouldn't get modified.

Isn't there a `known_first_party` flag for ruff as for isort? see our `pyproject.toml`

[the remark this answers is not in the document]
Comment (Borda) [reply examples/app/dag/app.py:10]: > I would say that ideally the tests imports wouldn't get modified.

yes if they were correct at first place

**Comment by Borda** [comment]
in fact the main difference is that you do not need to list fist  party packages but they are determined bu the folder names :flamingo:

**Review by justusschock — APPROVED** [review]

**Comment by Borda** [comment]
the App's failing is unrelated and it is addressed in #18627

**Review by lantiga — APPROVED** [review]
LGTM, did you triple-check the emphasize-lines in docs?

**Comment by Borda** [comment]
> LGTM, did you triple-check the emphasize-lines in docs?

revisited and fixed one, the rest was fine :)
